### PR TITLE
finished logs functionality and tests

### LINKED ISSUE #92: Add a minikube logs command to fetch all logs from the VM
This should support all relevant logs from inside the VM:
localkube, docker, etc.

Users should only run this to help debug issues.

### PR DESCRIPTION
**Pull request by aaron-prindle**
This adds a 'logs' command to minikube for debugging
Closes #92

### CODE REVIEW

File: cmd/minikube/cmd/logs.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/spf13/cobra"
+	"k8s.io/minikube/pkg/minikube/cluster"
+	"k8s.io/minikube/pkg/minikube/constants"
+)
+
+// logsCmd represents the logs command
+var logsCmd = &cobra.Command{
+	Use:   "logs",
+	Short: "Gets the logs of a local kubernetes cluster.",
+	Long:  `Gets the logs of a local kubernetes cluster.`,
```
Comment (dlorenc) [inline cmd/minikube/cmd/logs.go]: Can you add some text about how this is the logs of the cluster itself, and is only used for debugging of minikube, not user code? I could see users getting confused by this and trying to use it to get their own logs.

File: pkg/minikube/cluster/cluster.go [hunk 2]
```diff
@@ -36,6 +36,12 @@ import (
 	"k8s.io/minikube/pkg/util"
 )
 
+const (
+	remotePath             = "/var/lib/localkube/certs"
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: I think this was a merge conflict, I moved that line to constants.go.

File: pkg/minikube/cluster/cluster.go [hunk 3]
```diff
@@ -272,3 +278,25 @@ func GetHostDockerEnv(api libmachine.API) (map[string]string, error) {
...
+	}
+	if !exists {
+		return dne, nil
+	}
+
+	host, err := api.Load(constants.MachineName)
+	if err != nil {
+		return "", err
+	}
+	s, err := host.RunSSHCommand(fmt.Sprintf("tail -n +1 %s %s", remoteLocalKubeErrPath, remoteLocalKubeOutPath))
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: Use your helper in commands.go?

File: pkg/minikube/cluster/commands.go [hunk 4]
```diff
@@ -1,6 +1,9 @@
 package cluster
 
+import "fmt"
+
 var startCommand = `
 # Run with nohup so it stays up. Redirect logs to useful places.
 PATH=/usr/local/sbin:$PATH nohup sudo /usr/local/bin/localkube start --generate-certs=false > /var/log/localkube.out 2> /var/log/localkube.err < /dev/null &
 `
+var logsCommand = fmt.Sprintf("tail -n +1 %s %s", remoteLocalKubeErrPath, remoteLocalKubeOutPath)
```
Comment (dlorenc) [inline pkg/minikube/cluster/commands.go]: You could put the constants right in here to make it clearer, once you delete the old code.

File: cmd/minikube/cmd/logs.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		if err != nil {
+			log.Println("Error getting machine logs:", err)
+			os.Exit(1)
+		}
+		fmt.Fprintln(os.Stdout, s)
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(logsCmd)
```
Comment (luxas) [inline cmd/minikube/cmd/logs.go]: I think you should attach the `--v` flag here. See how to do that in the cobra documentation

File: pkg/minikube/cluster/cluster.go [hunk 6]
```diff
@@ -272,3 +278,25 @@ func GetHostDockerEnv(api libmachine.API) (map[string]string, error) {
 	}
 	return envMap, nil
 }
+
+// GetHostLogs gets the localkube logs of the host VM.
+func GetHostLogs(api libmachine.API) (string, error) {
+	dne := "Does Not Exist"
+	exists, err := api.Exists(constants.MachineName)
```
Comment (luxas) [inline pkg/minikube/cluster/cluster.go]: This code (to check if the machine exists, and if it does, try to load the machine from the api, etc...) seems generic enough for every minikube operation, so I think that you should break that out to a separate function that's used everywhere

**Comment by luxas** [comment]
This is a good start, but in the future, the user should be able to specify verbosity for the Kubernetes servers in `minikube start`, and `glog` should be used in `localkube`

File: cmd/minikube/cmd/logs.go (hunk 5 above)
Comment (luxas) [inline cmd/minikube/cmd/logs.go]: Oops, I meant it should be attached to the `start` command

Comment (dlorenc) [inline cmd/minikube/cmd/logs.go]: Do you mean for glog?

Comment (luxas) [inline cmd/minikube/cmd/logs.go]: Yes, so we may set verbosity for the k8s servers in localkube (and logging in localkube itself)

File: pkg/minikube/cluster/commands.go [hunk 7]
```diff
@@ -16,7 +16,15 @@ limitations under the License.
 
 package cluster
 
+import "fmt"
+
+const (
+	remoteLocalKubeErrPath = "/var/log/localkube.err"
```
Comment (luxas) [inline pkg/minikube/cluster/commands.go]: I don't really see the purpose of these constants if they aren't used where they should be used.
They are still hard-coded in startCommand.

**Comment by luxas** [comment]
@aaron-prindle Please rebase so we can get this in.

**Comment by luxas** [comment]
LGTM, let's get this in.
Again, letting @dlorenc make the merge as it's Aaron that has made it.

**Pull request merged**